### ISSUE TITLE
endoapply for Pairs Object

### ISSUE THREAD

**Issue by DarioS** [Bioconductor/S4Vectors#22]
If iterating through a Pairs object, `do.call` is currently necessary to keep it as a Pairs object.

```r
do.call(c, lapply(genePairs, function(pair)
{
  if(sample(c(FALSE, TRUE), 1)) # Reverse first and second gene.
    pair[2:1]
  else # Keep the order.
    pair
}))
```

Would providing a `endoapply` method for a `Pairs` object be useful and make the example code shown neater?

**Comment by lawremi** [comment]
I guess we would need a coercion from "List" and "list" to "Pairs". This would basically call `zipdown()`. Would you be up for submitting a PR?

**Comment by DarioS** [comment]
That seems to be it. The command `ans <- as(from, to_class, strict=FALSE)` in `coerce2` has an error `no method or default for coercing “list” to “Pairs”`. However, it seems that this part of the package will change substantially in the future

```
### TODO: Should probably use coercerToClass() internally (but coercerToClass()
### would first need to be improved).
```

so I shall leave it alone (also, perhaps coerceToClass or classCoercer is a grammatically better name for the function).

**Comment by lawremi** [comment]
The `coerce2()` stuff is independent of the need for the List/list to Pairs coercion. We just need a coerce method that calls `zipdown()`. This is a good opportunity for making a contribution.

**Comment by DarioS** [comment]
Already merged #25.
